study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/AddStateOption.java

```diff
@@ -0,0 +1,85 @@
+package org.broadinstitute.ddp.studybuilder.task;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import com.google.gson.Gson;
+import com.typesafe.config.Config;
+import org.broadinstitute.ddp.db.dao.JdbiActivityVersion;
+import org.broadinstitute.ddp.db.dao.PicklistQuestionDao;
+import org.broadinstitute.ddp.db.dto.ActivityVersionDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistOptionDef;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AddStateOption implements CustomTask {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AddStateOption.class);
+    private static final String[] studyGuids = {"cmi-pancan", "CMI-OSTEO", "cmi-brain", "rarex"};
+    private static final int DISPLAY_ORDER = 75; //between DE and FL
+    private String optionGson = "{\"allowDetails\":false,\"detailLabelTemplate\":null,\"exclusive\":false,"
+            + "\"optionLabelTemplate\":{\"templateCode\":null,\"templateText\":\"$STATE_dc\",\"templateType\":\"TEXT\",\"variables\":"
+            + "[{\"name\":\"STATE_dc\",\"translations\":[{\"language\":\"en\",\"text\":\"District of Columbia\"}]}]},\"stableId\":\"DC\"}";
+
+    private Path cfgPath;
+    private Config cfg;
+    private Config varsCfg;
+
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        this.cfgPath = cfgPath;
+        this.cfg = studyCfg;
+        this.varsCfg = varsCfg;
+    }
+
+    @Override
+    public void run(Handle handle) {
+
+        PicklistQuestionDao plQuestionDao = handle.attach(PicklistQuestionDao.class);
+
+        SqlHelper helper = handle.attach(SqlHelper.class);
+        for (String studyGuid : studyGuids) {
+            List<Long> questionIds = helper.findStatePLQuestionIdsByStudyGuid(studyGuid);
```

Comment:
As I understand it does the fix in several studies? Is it possible that same stableId ('STATE',  'SELF_STATE', 'CHILD_STATE', 'PARTICIPANT_STATE') has some question which not related to US states?

Reply:
Seems no such questions (other than with US states)..